### ISSUE TITLE
ngx.worker.id()  is always 0  on windows

### ISSUE THREAD

**Issue by 156704883** [openresty/openresty#721]
openresty version 1.19.3.1
nginx.conf worker_processes 4;
init_worker_by_lua_file resty_test/init.lua;
init.lua
ngx.log(ngx.ERR, "init worker: ",ngx.worker.id()) -- It's ngx.worker.id() always zero here

在winddows 下测试，ngx.worker.id() 得到的结果始终是0

**Comment by aifeiasdf** [comment]
because ngx.worker.id  implemented by fork() , so~

**Comment by aifeiasdf** [comment]
> openresty version 1.19.3.1
> nginx.conf worker_processes 4;
> init_worker_by_lua_file resty_test/init.lua;
> init.lua
> ngx.log(ngx.ERR, "init worker: ",ngx.worker.id()) -- It's ngx.worker.id() always zero here
> 
> 在winddows 下测试，ngx.worker.id() 得到的结果始终是0

because ngx.worker.id implemented by fork() , so~

**Comment by 156704883** [comment]
Yes, nginx is not suitable for production under windows, and I don't know when nginx will support IOCP. Because of various problems, I think I'll still use Linux instead of cross platform. Thank you for your reply



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年5月20日(星期四) 上午10:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [openresty/openresty] ngx.worker.id()  is always 0  on windows (#721)





  
openresty version 1.19.3.1
 nginx.conf worker_processes 4;
 init_worker_by_lua_file resty_test/init.lua;
 init.lua
 ngx.log(ngx.ERR, "init worker: ",ngx.worker.id()) -- It's ngx.worker.id() always zero here
 
在winddows 下测试，ngx.worker.id() 得到的结果始终是0
  
because ngx.worker.id implemented by fork() , so~
